src/modules/client/client.controller.ts

```diff
@@ -0,0 +1,102 @@
+import { 
+  Body,
+  Controller, 
+  Delete, 
+  Get, 
+  Param, 
+  ParseUUIDPipe, 
+  Post, 
+  Put, 
+  Query, 
+  Req, 
+  UseGuards
+} from '@nestjs/common';
+import {
+  ApiBadRequestResponse,
+  ApiBearerAuth,
+  ApiOkResponse,
+  ApiOperation,
+  ApiTags,
+  ApiUnauthorizedResponse,
+} from '@nestjs/swagger';
+import { JwtAuthGuard } from '../auth/jwt-auth.guard';
+import { Client } from '../../../entities/Client.entity';
+import { Request } from 'express';
+import { ClientService } from './client.service';
+import { DeleteResult } from 'typeorm';
+import { ParseOrderByPipeClients, ParseOrderPipeClients } from './client.pipes';
+import { PageResponse } from 'src/constants/PageResponse';
+
+@Controller('/client')
+@ApiTags('Clients')
+@ApiBearerAuth()
+@UseGuards(JwtAuthGuard)
+@ApiUnauthorizedResponse({ description: 'Unauthorized User' })
+@ApiBadRequestResponse({ description: 'Bad Request' })
+export class ClientController {
+  constructor(private readonly clientService: ClientService) {}
+
+  @Get('/:clientId')
+  @ApiOperation({summary: 'Retrieve details about an existing Client'})
+  @ApiOkResponse({ description: 'All info of a User', type: Client })
+  async getClientInfoById(
+    @Param('clientId', ParseUUIDPipe) clientId: string
+  ): Promise<Client> {
+    return this.clientService.getClientInfoById(clientId);
+  }
+
+  @Get()
+  @ApiOperation({ summary: 'Get all info of all Users based on its Id' })
+  @ApiOkResponse({ description: 'All info of all Users', type: Client })
```

Comment:
@fernandogcabal please update the summary and description, your are mentioning **Users** entity and the correct one will be **Client** entituy